### PR TITLE
feat: hide/show password

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RWDO-217

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/SetPassword/SetPasswordPresenter.swift [hunk 1]
```diff
@@ -28,8 +28,8 @@ final class SetPasswordPresenter: NSObject, Presenter, SetPasswordActionResponse
         ]
         
         let sectionRows: [Models.Section: [any Hashable]] = [
-            .password: [TextFieldModel(title: L10n.Account.enterPassword, value: item.password)],
-            .confirmPassword: [TextFieldModel(title: L10n.Account.confirmPassword, value: item.password)],
+            .password: [TextFieldModel(title: L10n.Account.enterPassword, value: item.password, trailing: .image(Asset.eyeShow.image))],
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SetPassword/SetPasswordPresenter.swift:31]: you don't need this. yo uare already doing it inside the view.

File: brd-ios/breadwallet/src/Redesign/Scenes/SetPassword/SetPasswordPresenter.swift [hunk 2]
```diff
@@ -28,8 +28,8 @@ final class SetPasswordPresenter: NSObject, Presenter, SetPasswordActionResponse
...
+            .confirmPassword: [TextFieldModel(title: L10n.Account.confirmPassword, value: item.password, trailing: .image(Asset.eyeShow.image))],
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SetPassword/SetPasswordPresenter.swift:32]: same

File: brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInPresenter.swift [hunk 3]
```diff
@@ -29,7 +29,7 @@ final class SignInPresenter: NSObject, Presenter, SignInActionResponses {
         
         let sectionRows: [Models.Section: [any Hashable]] = [
             .email: [TextFieldModel(title: L10n.Account.enterEmail, value: item.email)],
-            .password: [TextFieldModel(title: L10n.Account.enterPassword, value: item.password)],
+            .password: [TextFieldModel(title: L10n.Account.enterPassword, value: item.password, trailing: .image(Asset.eyeShow.image))],
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignIn/SignInPresenter.swift:32]: same

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift [hunk 4]
```diff
@@ -33,8 +33,8 @@ final class SignUpPresenter: NSObject, Presenter, SignUpActionResponses {
         
         let sectionRows: [Models.Section: [any Hashable]] = [
             .email: [TextFieldModel(title: L10n.Account.enterEmail, value: item.email)],
-            .password: [TextFieldModel(title: L10n.Account.createPassword, value: item.password)],
-            .confirmPassword: [TextFieldModel(title: L10n.Account.confirmPassword, value: item.password)],
+            .password: [TextFieldModel(title: L10n.Account.createPassword, value: item.password, trailing: .image(Asset.eyeShow.image))],
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:36]: same

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift [hunk 5]
```diff
@@ -33,8 +33,8 @@ final class SignUpPresenter: NSObject, Presenter, SignUpActionResponses {
...
+            .confirmPassword: [TextFieldModel(title: L10n.Account.confirmPassword, value: item.password, trailing: .image(Asset.eyeShow.image))],
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:37]: same

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 6]
```diff
@@ -248,9 +253,6 @@ class FETextField: FEView<TextFieldConfiguration, TextFieldModel>, UITextFieldDe
         
         trailingView.setup(with: viewModel.trailing)
         trailingView.isHidden = viewModel.trailing == nil
-        trailingView.snp.updateConstraints { make in
-            make.width.equalTo(viewModel.trailing == nil ? 0 : ViewSizes.extraSmall.rawValue)
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:252]: the issue with this is, the text will be under this view if it is too long. maybe change the alpha of that view?

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 7]
```diff
@@ -348,4 +356,9 @@ class FETextField: FEView<TextFieldConfiguration, TextFieldModel>, UITextFieldDe
     func changeToFirstResponder() {
         startEditing()
     }
+    
+    private func setupTogglableSecureEntry() {
+        let eye = textField.isSecureTextEntry == true ? Asset.eyeShow.image : Asset.eyeHide.image
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:361]: == true not needed here

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 8]
```diff
@@ -348,4 +356,9 @@ class FETextField: FEView<TextFieldConfiguration, TextFieldModel>, UITextFieldDe
...
+        trailingView.setup(with: .image(eye))
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:362]: can't you just edit the current config and model and call conffigure and setup again?

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 9]
```diff
@@ -288,6 +290,12 @@ class FETextField: FEView<TextFieldConfiguration, TextFieldModel>, UITextFieldDe
     }
     
     @objc private func trailingViewTapped() {
+        guard config?.isSecureTextEntry == false else {
+            textField.isSecureTextEntry.toggle()
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:294]: can't you toggle this in config?

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 10]
```diff
@@ -204,6 +204,11 @@ class FETextField: FEView<TextFieldConfiguration, TextFieldModel>, UITextFieldDe
         textField.autocorrectionType = config.autocorrectionType
         textField.keyboardType = config.keyboardType
         textField.isSecureTextEntry = config.isSecureTextEntry
+        if config.isSecureTextEntry {
+            trailingView.isHidden = false
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:208]: same here. can't you call setup again?

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift (hunk 6 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:252]: It's not going under, I checked

File: brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift [hunk 11]
```diff
@@ -69,17 +69,15 @@ final class SignUpPresenter: NSObject, Presenter, SignUpActionResponses {
                       passwordModel: .init(title: L10n.Account.createPassword,
                                            hint: !actionResponse.passwordsMatch && !actionResponse.isPasswordEmpty
                                            && !actionResponse.isPasswordAgainEmpty ? L10n.Account.passwordDoNotMatch : nil,
-                                           trailing: !actionResponse.passwordsMatch && !actionResponse.isPasswordEmpty
-                                           && !actionResponse.isPasswordAgainEmpty ? .image(Asset.warning.image.tinted(with: LightColors.Error.one)) : nil,
-                                           displayState: actionResponse.passwordState),
+                                           displayState: actionResponse.passwordState,
+                                           showPasswordToggle: true),
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/SignUp/SignUpPresenter.swift:73]: don't you already have isSecureTextEntry?

**Review by kenanmamedoff — APPROVED** [review]

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**